scRNA-seq/post-processing_scRNA-seq.Rmd

```diff
@@ -0,0 +1,167 @@
+look as good as the full set.
+```{r alevinQC}
+# Produce a QC report
+alevinQC::alevinQCReport(alevin_file,
+                         sampleId = "pbmc_1k_v2_S1_L001_subset", 
+                         outputFile = "pbmc_1k_v2_S1_L001_qc_report.html", 
+                         outputDir = "data",
+                         outputFormat = "html_document")
+```
+
+#### Import counts from Alevin output
+Here we are using the function we set up in the earlier chunk. 
+```{r Import Data}
+# Read in the data
+alv_data <- ReadAlevin(alevin_file)
+```
+
+#### Set up the Seurat object and apply a filter to the data. 
+In this example, we are applying a gene filter cutoff of at least 3 cells 
+expressing each gene. 
+For cell filtering, each cell must express at least 200 genes.
+```{r Set Up Seurat Object}
+# Set up the seurat object 
+seurat_obj <- CreateSeuratObject(raw.data = alv_data, min.cells = 3,
+                                 min.genes = 200, 
+                                 project = "1k_pbmc_subset")
+```
+
+#### Normalize the data using log normalization
+These are the default settings for normalization and should work in most 
+instances. 
+```{r Normalize the Data}
+seurat_norm <- NormalizeData(object = seurat_obj, 
+                             normalization.method = "LogNormalize",
+                             scale.factor = 10000)
+```
+
+#### Identify variable genes and scale the data
+In this step, we will identify genes with higher variance. 
+These are probably the genes that are of more interest to us and they are what 
+we will use for Principal Components Analysis in the upcoming step.
+```{r Find Variable Genes}
+seurat_norm <- FindVariableGenes(object = seurat_norm, mean.function = ExpMean,
+                                x.low.cutoff = 0.0125, x.high.cutoff = 3,
+                                y.cutoff = 0.5, do.plot = FALSE)
+
+# Scale the data
+seurat_norm <- ScaleData(object = seurat_norm)
+```
+
+#### Execute PCA with high variance genes
+Just like we did with bulk RNA-seq, we will also run PCA with our single cell data. 
+However, because single cell data is more skewed as far as gene expression, we 
+will do our PCA based on the high variance genes, which should give us a more 
+informative analysis.
+```{r PCA}
+# Run PCA
+seurat_norm <- RunPCA(object = seurat_norm, pc.genes = seurat_norm@var.genes,
+                      do.print = TRUE, pcs.print = 1:5, genes.print = 5)
+
+# Cluster analyses
+seurat_norm <- FindClusters(object = seurat_norm, reduction.type = "pca", 
+                            dims.use = 1:10, resolution = 0.6, print.output = 0, 
+                            save.SNN = TRUE)
+```
+
+#### Run tSNE on normalized data
+```{r Run tSNE}
+# Run tSNE 
+seurat_norm <- RunTSNE(object = seurat_norm, dims.use = 1:10, do.fast = TRUE)
+
+# Plot the tSNE
+TSNEPlot(object = seurat_norm, colors.use = RColorBrewer::brewer.pal(9, "Greys"))
```

Comment:
Also probably for when #27 gets tackled -- should get @dvenprasad to weigh in on the palette accessibility.

Reply:
This ends up being completely changed in my upcoming PR that addresses #27